troposphere/haze/source/ptp_object_database.cpp

```diff
@@ -108,4 +142,13 @@ namespace haze {
         return std::addressof(*it);
     }
 
+    PtpObject *PtpObjectDatabase::GetObjectByName(const char *name) {
+        /* Find in name mapping. */
+        auto it = m_name_to_object_id.find_key(name);
```

Comment:
This code is completely fine and you do not need to change it.

For your ams style reference, the way I would write it:

```cpp
    PtpObject *PtpObjectDatabase::GetObjectByName(const char *name) {
        /* Find in name mapping. */
        if (auto it = m_name_to_object_id.find_key(name); it != m_name_to_object_id.end()) {
            return std::addressof(*it);
        } else {
            return nullptr;
        }
    }
```